src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/CollectionPropertiesShouldBeReadOnlyTests.cs

```diff
@@ -431,5 +431,39 @@ class A
 }
 ");
         }
+
+        [Fact, WorkItem(4461, "https://github.com/dotnet/roslyn-analyzers/issues/4461")]
+        public async Task CA2227_CSharp_InitPropertyRecord()
+        {
+            await new VerifyCS.Test
+            {
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.CSharp9,
+                TestCode = @"
+using System.Collections.Generic;
+
+public record MyRecord(IList<int> Items);",
```

Comment:
@Evangelink For the issue you're facing in https://github.com/dotnet/roslyn-analyzers/pull/4463/files#r531133849

I think you can fix it like this:

```suggestion
public record MyRecord(IList<int> Items);

namespace System.Runtime.CompilerServices
{
    class IsExternalInit {}
}",
```

Reply:
The tests of the IDE-side of roslyn currently follow that approach:

See https://github.com/dotnet/roslyn/blob/cf55f3a58e47298426fa971d3bd9d8857c746c65/src/Analyzers/CSharp/Tests/AddAccessibilityModifiers/AddAccessibilityModifiersTests.cs#L163

Reply:
It should work but my understanding was that we try not to rely on fake class and instead use the real code.

Reply:
@Evangelink The real class is already empty.

https://github.com/dotnet/runtime/blob/6072e4d3a7a2a1493f514cdf4be75a3d56580e84/src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/IsExternalInit.cs

Reply:
I meant we replaced the various fake class in the tests of roslyn-analyzer to rely on having a reference to the real dll (direct dll or nuget package). I think here we should also do the same.

Reply:
@Evangelink I'm not sure what is the thing that determines the runtime version under which the tests are run.
Could updating MicrosoftCodeAnalysisTestingVersion help? Not sure.